In [3]:
pip install git+git://github.com/datamade/census.git

  Cloning git://github.com/datamade/census.git to c:\users\asmita\appdata\local\temp\pip-req-build-1s_aaov8
  Resolved git://github.com/datamade/census.git to commit 70e2c08710c1e10e5bc2054b78613fa8794d4765
  Installing build dependencies: started
Note: you may need to restart the kernel to use updated packages.


  Running command git clone -q git://github.com/datamade/census.git 'C:\Users\asmita\AppData\Local\Temp\pip-req-build-1s_aaov8'


  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
  Created wheel for census: filename=census-0.8.18-py3-none-any.whl size=11110 sha256=21123caf5c4b7774ab7effaf9cdacbc08f886138d27bf95e77845de8bf2f4a38
  Stored in directory: C:\Users\asmita\AppData\Local\Temp\pip-ephem-wheel-cache-49m76fyv\wheels\20\be\40\0e0dc91af2e1fd78c99618924a422c92a587189953a2903427
Successfully built census


In [11]:
pip install census

In [6]:
pip install us

  Using cached us-2.0.2.tar.gz (14 kB)
  Using cached jellyfish-0.6.1.tar.gz (132 kB)
  Created wheel for us: filename=us-2.0.2-py3-none-any.whl size=11945 sha256=2759425c1317d5c2b9a27fb262f7a499f58aa3041108d0b8b5d03a0f38e7cf9e
  Stored in directory: c:\users\asmita\appdata\local\pip\cache\wheels\07\4c\fa\a65ed0f9c00360e785327fc370eba55ca575cc3508ab13ed8e
  Created wheel for jellyfish: filename=jellyfish-0.6.1-cp38-cp38-win_amd64.whl size=26589 sha256=b6cd26a480ea1fb58eecf41bad2f8663498e17f55486a634cf3bebad5958cc87
  Stored in directory: c:\users\asmita\appdata\local\pip\cache\wheels\aa\a9\a6\359d736dbeda356e6ab8a7ff70a3db636cefe91df7d09f63dc
Successfully built us jellyfish
Note: you may need to restart the kernel to use updated packages.


In [15]:
from census import Census
from us import states

In [44]:
#MY_API_KEY = "b03aa3d496e48011cf18dfeee3f411cfa6f1204d"
MY_API_KEY = "a8aa21718e908df84ccc732b10ba0394d273f3dd"

In [42]:
c = Census(MY_API_KEY)

In [45]:
c.acs5.get(('NAME', 'B25034_010E'),
          {'for': 'state:{}'.format(states.MD.fips)})

[{'NAME': 'Maryland', 'B25034_010E': 128476.0, 'state': '24'}]

Let's try to look at some relevant tables. First, let's consider the "Median Age At First Marriage" table. From censusreporter.org, we find that this is Table B12007. Let's try looking at this table in the default year, 2019.  

Note: I was confused about how to find the table's suffix (e.g., the \_010E in the table above), and I stumbled upon this [https://api.census.gov/data/2019/acs/acs5/groups/B12007.html]. I still don't know how exactly to find the suffixes, though. 

In [46]:
import pandas as pd

median_age_male = pd.DataFrame(c.acs5.get(('NAME', 'B12007_001E'), {'for': 'state:*'}))
median_age_male

NAME  B12007_001E state
0                Alabama         28.5    01
1                 Alaska         29.2    02
2                Arizona         29.9    04
3               Arkansas         27.2    05
4             California         30.8    06
5               Colorado         29.7    08
6               Delaware         30.4    10
7   District of Columbia         30.9    11
8            Connecticut         31.1    09
9                Florida         30.7    12
10               Georgia         29.5    13
11                 Idaho         27.0    16
12                Hawaii         29.3    15
13              Illinois         30.5    17
14               Indiana         29.0    18
15                  Iowa         28.4    19
16                Kansas         27.8    20
17              Kentucky         28.1    21
18             Louisiana         29.7    22
19                 Maine         29.7    23
20              Maryland         30.5    24
21         Massachusetts         31.0    25
22              Michigan         30.0    26
23             Minnesota         29.7    27
24           Mississippi         28.8    28
25              Missouri         28.5    29
26               Montana         29.1    30
27              Nebraska         28.2    31
28                Nevada         29.8    32
29         New Hampshire         30.5    33
30            New Jersey         30.7    34
31            New Mexico         29.3    35
32              New York         31.1    36
33        North Carolina         29.3    37
34          North Dakota         28.2    38
35                  Ohio         29.6    39
36              Oklahoma         27.6    40
37                Oregon         29.9    41
38          Pennsylvania         30.3    42
39          Rhode Island         30.9    44
40        South Carolina         29.6    45
41          South Dakota         27.8    46
42             Tennessee         28.7    47
43                 Texas         28.9    48
44               Vermont         30.6    50
45                  Utah         26.4    49
46              Virginia         29.7    51
47            Washington         29.5    53
48         West Virginia         28.7    54
49             Wisconsin         29.6    55
50               Wyoming         28.0    56
51           Puerto Rico         32.4    72

In [47]:
# cleaning up the table
median_age_male["AGE"] = median_age_male["B12007_001E"]
median_age_male = median_age_male[["NAME", "AGE"]]
median_age_male

NAME   AGE
0                Alabama  28.5
1                 Alaska  29.2
2                Arizona  29.9
3               Arkansas  27.2
4             California  30.8
5               Colorado  29.7
6               Delaware  30.4
7   District of Columbia  30.9
8            Connecticut  31.1
9                Florida  30.7
10               Georgia  29.5
11                 Idaho  27.0
12                Hawaii  29.3
13              Illinois  30.5
14               Indiana  29.0
15                  Iowa  28.4
16                Kansas  27.8
17              Kentucky  28.1
18             Louisiana  29.7
19                 Maine  29.7
20              Maryland  30.5
21         Massachusetts  31.0
22              Michigan  30.0
23             Minnesota  29.7
24           Mississippi  28.8
25              Missouri  28.5
26               Montana  29.1
27              Nebraska  28.2
28                Nevada  29.8
29         New Hampshire  30.5
30            New Jersey  30.7
31            New Mexico  29.3
32              New York  31.1
33        North Carolina  29.3
34          North Dakota  28.2
35                  Ohio  29.6
36              Oklahoma  27.6
37                Oregon  29.9
38          Pennsylvania  30.3
39          Rhode Island  30.9
40        South Carolina  29.6
41          South Dakota  27.8
42             Tennessee  28.7
43                 Texas  28.9
44               Vermont  30.6
45                  Utah  26.4
46              Virginia  29.7
47            Washington  29.5
48         West Virginia  28.7
49             Wisconsin  29.6
50               Wyoming  28.0
51           Puerto Rico  32.4

In [48]:
median_age_male["AGE"].mean()

29.42884615384616

In [49]:
median_age_female = pd.DataFrame(c.acs5.get(('NAME', 'B12007_002E'), {'for': 'state:*'}))
median_age_female

NAME  B12007_002E state
0                Alabama         26.7    01
1                 Alaska         26.4    02
2                Arizona         27.8    04
3               Arkansas         25.7    05
4             California         29.0    06
5               Colorado         27.7    08
6               Delaware         29.0    10
7   District of Columbia         30.6    11
8            Connecticut         29.5    09
9                Florida         28.9    12
10               Georgia         27.7    13
11                 Idaho         25.4    16
12                Hawaii         28.4    15
13              Illinois         28.9    17
14               Indiana         27.0    18
15                  Iowa         26.6    19
16                Kansas         26.1    20
17              Kentucky         26.4    21
18             Louisiana         27.9    22
19                 Maine         27.8    23
20              Maryland         29.1    24
21         Massachusetts         29.9    25
22              Michigan         28.1    26
23             Minnesota         27.8    27
24           Mississippi         26.9    28
25              Missouri         26.9    29
26               Montana         26.4    30
27              Nebraska         26.4    31
28                Nevada         27.8    32
29         New Hampshire         28.7    33
30            New Jersey         29.3    34
31            New Mexico         27.7    35
32              New York         29.7    36
33        North Carolina         27.6    37
34          North Dakota         26.0    38
35                  Ohio         27.9    39
36              Oklahoma         25.9    40
37                Oregon         27.9    41
38          Pennsylvania         28.7    42
39          Rhode Island         29.7    44
40        South Carolina         28.0    45
41          South Dakota         26.4    46
42             Tennessee         26.9    47
43                 Texas         27.0    48
44               Vermont         28.4    50
45                  Utah         24.6    49
46              Virginia         27.8    51
47            Washington         27.3    53
48         West Virginia         26.7    54
49             Wisconsin         27.8    55
50               Wyoming         25.7    56
51           Puerto Rico         30.8    72

In [50]:
# cleaning up the table
median_age_female["AGE"] = median_age_female["B12007_002E"]
median_age_female = median_age_female[["NAME", "AGE"]]

median_age_female["AGE"].mean()

27.67884615384616

In [56]:
median_age_male_2010 = pd.DataFrame(c.acs5.get(('NAME', 'B12007_001E'), {'for': 'state:*'}, year = 2010))
median_age_male_2010

NAME  B12007_001E state
0                Alabama         27.0    01
1                 Alaska         27.1    02
2                Arizona         27.4    04
3               Arkansas         25.9    05
4             California         29.0    06
5               Colorado         27.3    08
6            Connecticut         29.4    09
7               Delaware         28.2    10
8   District of Columbia         30.8    11
9                Florida         28.7    12
10               Georgia         27.8    13
11                Hawaii         28.6    15
12                 Idaho         25.4    16
13              Illinois         28.8    17
14               Indiana         27.1    18
15                  Iowa         27.1    19
16                Kansas         26.6    20
17              Kentucky         26.7    21
18             Louisiana         27.7    22
19                 Maine         28.0    23
20              Maryland         29.1    24
21         Massachusetts         29.8    25
22              Michigan         28.4    26
23             Minnesota         27.9    27
24           Mississippi         27.1    28
25              Missouri         27.0    29
26               Montana         27.7    30
27              Nebraska         26.9    31
28                Nevada         27.9    32
29         New Hampshire         28.9    33
30            New Jersey         29.8    34
31            New Mexico         28.0    35
32              New York         29.9    36
33        North Carolina         27.6    37
34          North Dakota         27.1    38
35                  Ohio         27.9    39
36              Oklahoma         26.0    40
37                Oregon         28.1    41
38          Pennsylvania         28.9    42
39          Rhode Island         29.6    44
40        South Carolina         28.0    45
41          South Dakota         26.5    46
42             Tennessee         26.9    47
43                 Texas         27.0    48
44                  Utah         25.2    49
45               Vermont         28.5    50
46              Virginia         28.1    51
47            Washington         27.8    53
48         West Virginia         27.0    54
49             Wisconsin         28.1    55
50               Wyoming         26.7    56
51           Puerto Rico         28.8    72

In [57]:
# cleaning up the table
median_age_male_2010["AGE"] = median_age_male_2010["B12007_001E"]
median_age_male_2010 = median_age_male_2010[["NAME", "AGE"]]
# calculating mean median age across states
median_age_male_2010["AGE"].mean()

27.82307692307692

In [60]:
median_age_female_2010 = pd.DataFrame(c.acs5.get(('NAME', 'B12007_002E'), {'for': 'state:*'}, year = 2010))
median_age_female_2010

NAME  B12007_002E state
0                Alabama         25.5    01
1                 Alaska         24.8    02
2                Arizona         25.6    04
3               Arkansas         24.5    05
4             California         26.9    06
5               Colorado         25.7    08
6            Connecticut         27.6    09
7               Delaware         26.4    10
8   District of Columbia         29.6    11
9                Florida         26.6    12
10               Georgia         25.9    13
11                Hawaii         27.0    15
12                 Idaho         23.3    16
13              Illinois         27.1    17
14               Indiana         25.9    18
15                  Iowa         25.5    19
16                Kansas         25.1    20
17              Kentucky         25.0    21
18             Louisiana         26.2    22
19                 Maine         26.5    23
20              Maryland         27.4    24
21         Massachusetts         28.5    25
22              Michigan         26.6    26
23             Minnesota         26.3    27
24           Mississippi         25.9    28
25              Missouri         25.8    29
26               Montana         25.4    30
27              Nebraska         25.5    31
28                Nevada         25.7    32
29         New Hampshire         26.8    33
30            New Jersey         27.8    34
31            New Mexico         25.9    35
32              New York         28.3    36
33        North Carolina         25.9    37
34          North Dakota         25.7    38
35                  Ohio         26.3    39
36              Oklahoma         24.5    40
37                Oregon         26.1    41
38          Pennsylvania         27.2    42
39          Rhode Island         27.9    44
40        South Carolina         26.4    45
41          South Dakota         25.5    46
42             Tennessee         25.4    47
43                 Texas         25.3    48
44                  Utah         23.0    49
45               Vermont         27.0    50
46              Virginia         26.5    51
47            Washington         25.8    53
48         West Virginia         25.4    54
49             Wisconsin         26.4    55
50               Wyoming         24.2    56
51           Puerto Rico         26.7    72

In [61]:
# cleaning up table
median_age_female_2010["AGE"] = median_age_female_2010["B12007_002E"]
median_age_female_2010 = median_age_female_2010[["NAME", "AGE"]]
# calculating mean median age across states
median_age_female_2010["AGE"].mean()

26.111538461538466

We see that in 2019, the median age of first marriage was 29.43 years for males and 27.68 years for females. In 2010, however, the male median age was 27.82, and the female median age was 26.11. In less than a decade, the median age for fir